### ISSUE TITLE
ios8: dismissing UIAlertView swallows calls to didDismissWithButtonIndex:

### ISSUE THREAD

**Issue by marcpowell-okta** [kif-framework/KIF#712]
Problem:
I'm dismissing a UIAlertView using KIF:

```
    [tester tapViewWithAccessibilityLabel:@"Ok"];
```

But my delegate method alertView:didDismissWithButtonIndex: is never called with this tap- so my UI state is out of whack.

Workaround was to move my state update into alertView:willDismissWithButtonIndex:, which does get called by the "Ok" tap.

**Comment by phatmann** [comment]
According to [this post](http://stackoverflow.com/questions/17536328/alertview-diddismisswithbuttonindex-never-called), this happens because the alert is dismissed immediately after showing it. If you add a little delay, does this problem go away?

**Comment by marcpowell-okta** [comment]
I added a 1s, then 3s, then 5s delay, and the problem didn't go away.
There is a background/foreground immediately before this behavior.
Using KIF 3.2.3+ from commit d2567a99a14414949fe8217211cda54a3e9f3f92 in Xcode 6.4

I verified that didDismissWithButtonIndex: does get called in the KIF Tests 3.2.3 from commit a022fd48c8b2187c1c6e2e904f16784749c53fc7 in Xcode 6.4 

Will try and isolate if I have time- leaving this issue here with a workaround in case anybody else encounters it.
